pallets/gear/src/internal.rs

```diff
@@ -631,7 +655,10 @@ where
             let hold = delay_hold.min(maximal_hold);
 
             // Locking funds for holding.
-            GasHandlerOf::<T>::lock(dispatch.id(), hold.lock())
+            let lock_id = hold.lock_id().unwrap_or_else(|| {
```

Comment:
Consider remove `lock_id` function as it only used in cases like this - `StorageType` is few lines above

Reply:
> Consider create `LockId`, then convert into `StorageType` to avoid these `unwrap_or_else`

That would invert the flow of reasoning. It's not the `LockId` that determines the `StorageType` which, in turn, determines the `HoldBound` but the other way around: we create a `HoldBound` for a specific `StorageType` which may (or may not in theory) have an associated `LockId`. In reality though any storage we are currently charging for would have a lock therefore this `_or_else` part will never execute.